modules/issue_tracker/ajax/EditIssue.php

```diff
@@ -0,0 +1,619 @@
+        array('issueID' => $issueID));
+
+    $whoIsWatching = array();
+    foreach ($watching as $watcher) {
+        $whoIsWatching[] = $watcher['userID'];
+    }
+    error_log(json_encode($whoIsWatching));
+    return $whoIsWatching;
+}
+
+/**
+ * Gets the changes to values, and the comments relevant to the given issue
+ *
+ * @param int $issueID the issueID
+ *
+ * @throws DatabaseException
+ *
+ * @return string $commentHistory html
+ */
+function getComments($issueID)
+{
+    $db =& Database::singleton();
+    $unformattedComments = $db->pselect(
+        "SELECT newValue, fieldChanged, dateAdded, addedBy " .
+        "FROM issues_history where issueID=:issueID " .
+        "UNION " .
+        "SELECT issueComment, 'comment', dateAdded, addedBy " .
+        "FROM issues_comments where issueID=:issueID " .
+        "ORDER BY dateAdded",
+        array('issueID' => $issueID)
+    );
+
+    //todo: get real names as well
+
+    $commentHistory = '';
+    foreach ($unformattedComments as $comment) {
+        $commentString = "";
+        $commentString .= '[' . $comment['dateAdded'] . '] ';
+        $commentString .= $comment['addedBy'];
+        if ($comment['fieldChanged'] === 'comment') {
+            $commentString .= ' commented ' . '<i>' . $comment['newValue'] . '</i>';
+        } else if ($comment['fieldChanged'] === 'module') {
+            $module = $db->pselectOne(
+                "SELECT Label FROM LorisMenu WHERE ID=:module",
+                array('module' => $comment['newValue'])
+            );
+            $commentString .= " updated the <b>" . $comment['fieldChanged'] .
+                "</b> to <i>" . $module . "</i>";
+        } else if ($comment['fieldChanged'] === 'centerID') {
+            $site = $db->pselectOne(
+                "SELECT Name FROM psc WHERE CenterID=:centerID",
+                array('centerID' => $comment['newValue'])
+            );
+            $commentString .= " updated the <b>" . "site" .
+                "</b> to <i>" . $site . "</i>";
+        } else if ($comment['fieldChanged'] === 'candID') {
+            $PSCID = $db->pselectOne(
+                "SELECT PSCID FROM candidate WHERE CandID=:candID",
+                array('candID' => $comment['newValue'])
+            );
+            $commentString .= " updated the <b>" . "PSCID" .
+                "</b> to <i>" . $PSCID . "</i>";
+        } else if ($comment['fieldChanged'] === 'sessionID') {
+            $visitLabel = $db->pselectOne(
+                "SELECT Visit_label FROM session WHERE ID=:sessionID",
+                array('sessionID' => $comment['newValue'])
+            );
+            $commentString .= " updated the <b>" . "visit label" .
+                "</b> to <i>" . $visitLabel . "</i>";
+        } else {
+            $commentString .= " updated the <b>" . $comment['fieldChanged'] .
+                "</b> to <i>" . $comment['newValue'] . "</i>";
+        }
+        $commentHistory .= $commentString . '<br/>';
```

Comment:
There shouldn't be HTML mixed in with PHP code. This whole thing would be a lot cleaner if it was in a smarty template that just gets rendered here. You can look at how instrument_list_controlpanel.tpl gets rendered in the instrument_list module for an example.